lib/address_bar.dart

```diff
@@ -38,11 +38,32 @@ class AddressBar extends StatelessWidget {
                           color: Colors.black,
                         ),
                         controller: controller,
-                        onSubmitted: (value) {
-                          var newURL = Uri.tryParse(value);
-                          if (!newURL.hasScheme) {
-                            newURL = Uri.tryParse("gemini://" + value);
+                        onSubmitted: (value) async {
+                          Uri newURL;
+                          if (validator.isURL(value, {
+                                "protocols": ['gemini'],
+                                "require_tld": true,
+                                "require_protocol": false,
+                                "allow_underscores": true
+                              }) ||
+                              validator.isURL(value, {
+                                "protocols": ['gemini'],
+                                "require_tld": false,
+                                "require_protocol": true,
+                                "allow_underscores": true
+                              })) {
+                            newURL = Uri.tryParse(value);
+                            if (!newURL.hasScheme) {
+                              newURL = Uri.parse("gemini://" + value);
+                            }
+                          } else {
+                            String searchEngine =
+                                (await SharedPreferences.getInstance())
+                                    .getString("search");
+                            newURL = Uri.parse(searchEngine);
```

Comment:
When I first ran and typed in a search this threw a null pointer exception, going into settings and pressing enter on the search field fixed it.

```
E/flutter (22250): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'length' was called on null.
E/flutter (22250): Receiver: null
E/flutter (22250): Tried calling: length
E/flutter (22250): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
E/flutter (22250): #1      Uri.parse (dart:core/uri.dart:792:17)
E/flutter (22250): #2      AddressBar.build.<anonymous closure>
package:deedum/address_bar.dart:63
E/flutter (22250): <asynchronous suspension>
E/flutter (22250): #3      AddressBar.build.<anonymous closure> (package:deedum/address_bar.dart)
package:deedum/address_bar.dart:1
E/flutter (22250): #4      EditableTextState._finalizeEditing
package:flutter/…/widgets/editable_text.dart:1819
E/flutter (22250): #5      EditableTextState.performAction
package:flutter/…/widgets/editable_text.dart:1688
E/flutter (22250): #6      TextInput._handleTextInputInvocation
package:flutter/…/services/text_input.dart:1168
E/flutter (22250): #7      MethodChannel._handleAsMethodCall
package:flutter/…/services/platform_channel.dart:430
E/flutter (22250): #8      MethodChannel.setMethodCallHandler.<anonymous closure>
package:flutter/…/services/platform_channel.dart:383
E/flutter (22250): #9      _DefaultBinaryMessenger.handlePlatformMessage
package:flutter/…/services/binding.dart:283
E/flutter (22250): #10     _invoke3.<anonymous closure> (dart:ui/hooks.dart:280:15)
```